gips/data/landsat/landsat.py

```diff
@@ -286,6 +289,53 @@ def __init__(self, filename):
             raise Exception("Sensor %s not supported: %s" % (self.sensor, filename))
         self._version = self.version
 
+    def filter(self, pclouds=100, **kwargs):
+        if pclouds < 100:
+            if os.path.exists(self.filename):
+                mtlfilename = [f for f in self.datafiles() if 'MTL.txt' in f][0]
+                mtlfilename = self.extract([mtlfilename])[0]
+                with utils.error_handler('Error reading metadata file ' + mtlfilename):
+                    with open(mtlfilename, 'r') as mtlfile:
+                        text = mtlfile.read()
+                    cloud_cover = re.match(
+                        r".*CLOUD_COVER = (\d+.?\d*)",
```

Comment:
Looks great!  :+1:  Optional:  But if you want, you can leave out the leading `.*` as it matches literally anything or nothing, so at the beginning of a pattern it doesn't add anything to the semantics of the regex.

Reply:
But it doesn't match if you don't have the `.*`:
```
>>> re.match('CLOUDCOVER.*', 'asdf jkl CLOUDCOVER')
>>> re.match('.*CLOUDCOVER.*', 'asdf jkl CLOUDCOVER')
<_sre.SRE_Match object at 0x7f2f4933a4a8>
```

Reply:
Oops, you're right, I don't think I knew that `re.match` has an implicit `^`:

https://docs.python.org/2/library/re.html#search-vs-match